test/test_nn.py

```diff
@@ -8999,6 +8999,11 @@ def test_BatchNorm_empty(self, device):
             with torch.backends.cudnn.flags(enabled=False):
                 self._test_module_empty_input(mod, inp)
 
+        self.assertEqual(mod.running_mean, torch.tensor([0., 0, 0], device=device))
+        self.assertEqual(mod.running_var, torch.tensor([1., 1, 1], device=device))
+        self.assertEqual(mod.weight.grad, None)
```

Comment:
mod.weight.grad should be zeros rather than None, although currently it's implemented as None. Is the goal of this PR to add it now so the tests can pass and change it later?

I added a comment in `_test_module_empty_input` that we could check `weight.grad` and `bias.grad` there.

Reply:
Alright. I will change the `batch_norm` behaviour to reflect the zeros.

Reply:
29fabb1fbcb6b62c9e5caf1391d66238d679c671 should have changed the gradient to zero now and include the tests. However it seems that the test for `running_mean` and `running_var` is still not included. That seems to be the only action item left for BatchNorm.